pkg/controllers/oauthclientscontroller/oauthclientscontroller.go

```diff
@@ -158,16 +158,24 @@ func randomBits(bits uint) []byte {
 	return b
 }
 
-func ensureOAuthClient(ctx context.Context, oauthClients oauthclient.OAuthClientInterface, client oauthv1.OAuthClient) error {
-	_, err := oauthClients.Create(ctx, &client, metav1.CreateOptions{})
-	if err == nil || !apierrors.IsAlreadyExists(err) {
+func ensureOAuthClient(ctx context.Context, oauthClients oauthclient.OAuthClientInterface, oauthClientLister oauthv1listers.OAuthClientLister, client oauthv1.OAuthClient) error {
+	_, err := oauthClientLister.Get(client.Name)
+	if apierrors.IsNotFound(err) {
+		_, err = oauthClients.Create(ctx, &client, metav1.CreateOptions{})
 		return err
+	} else if err != nil {
+		if _, err = oauthClients.Get(ctx, client.Name, metav1.GetOptions{}); err != nil {
+			return err
+		}
 	}
```

Comment:
Same question here, what error state from the lister will result in us getting a better result from a live GET?